README.md

```diff
@@ -1,8 +1,194 @@
+
+## Development Docker ##
+
+The docker images used for aurora development are [here](https://cloud.docker.com/u/shadowrobot/repository/docker/shadowrobot/aurora-molecule-devel)
+
+Currently both xenial and bionic tags are working well and there is no difference.
+The rest of the document assumes the user is using bionic tag.
+
+Instructions on how to use this:
+1. Use Ubuntu 16.04 or Ubuntu 18.04 computer
+
+2. Install Docker (using instructions from [here](https://docs.docker.com/install/linux/docker-ce/ubuntu/))
+
+3. Run the following command in terminal to create a container for aurora development:
+
+```
+docker run -it --name aurora_dev -e DISPLAY -e QT_X11_NO_MITSHM=1 -e LOCAL_USER_ID=$(id -u) -v /var/run/docker.sock:/var/run/docker.sock -v /tmp/.X11-unix:/tmp/.X11-unix:rw shadowrobot/aurora-molecule-devel:bionic
+```
+4. Once the container has launched, clone aurora to it:
+```
+git clone https://github.com/shadow-robot/aurora.git
+```
+5. Go into the aurora folder:
+```
+cd aurora
+```
+6. Open Visual Studio Code which is already installed inside the Docker container:
+```
+code
+```
+7. Open the aurora folder in Visual Studio Code
+
+# Testing #
+
+## Testing with molecule_docker ##
+
+Once you have written your code for aurora in your branch, test it locally with Molecule first before pushing to GitHub.
+
+1. In the docker container terminal go to /ansible/playbooks/molecule_docker folder
+
+```
+cd /home/user/aurora/ansible/playbooks/molecule_docker
+```
+
+2. Run the following command to execute all molecule tests locally with debug mode:
+
+```
+molecule --debug test --all
+```
+
+3. Fix any errors
+
+4. If you want to run a specific test case (scenarios are in ansible/playbooks/molecule_docker/molecule folder), use:
+
+```
+molecule --debug test -s name_of_your_scenario
+```
+5. Often it is useful to run molecule in stages (create, converge, verify, login (if necessary), and finally destory) for better debugging (so you can inspect every stage yourself). See [this](https://molecule.readthedocs.io/en/stable/usage.html) page, and do, for example:
+```
+molecule --debug create -s name_of_your_scenario
+molecule --debug converge -s name_of_your_scenario
+molecule --debug verify -s name_of_your_scenario
+molecule --degub login -s name_of_your_scenario
+molecule --debug destroy -s name_of_your_scenario
+```
+## Private docker images ##
+
+At the moment, we don't want to give molecule access to private docker hub credentials for private docker images (e.g. shadow-teleop). That is why, in every playbook.yml inside the test scenarios in the molecule_docker folder, we override the image with image="shadowrobot/dexterous-hand" for any teleop-related test scenario. When we actually deploy Aurora, the user will be asked to fill in their private Docker hub credentials.
```

Comment:
Not super clear. Each test we have uses image="shadowrobot/dexterous-hand" by default and if you want to run the tests for another scenario we need to change this and then be asked for docker hub credentials?